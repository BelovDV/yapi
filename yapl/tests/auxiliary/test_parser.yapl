fn func() -> (T)
      fn <-- is simple
  (("hello world \t \n \\ \" is ok? ".right))
        extend (doing [1 + 2345])
  return

.fn test_set()
  { gen 3 }.map(?).fold(+) ?
    .read .int
