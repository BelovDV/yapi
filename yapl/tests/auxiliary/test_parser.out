[
    Line {
        sent: Sent {
            chains: [
                Chain {
                    chain: [
                        Expr {
                            expr: Symbol(
                                Symbol(0: "fn"),
                            ),
                            span: Span(0, 2),
                        },
                    ],
                    span: Span(0, 2),
                },
                Chain {
                    chain: [
                        Expr {
                            expr: Symbol(
                                Symbol(1: "func"),
                            ),
                            span: Span(3, 7),
                        },
                        Expr {
                            expr: Bracket {
                                ty: Round,
                                inner: [],
                            },
                            span: Span(7, 9),
                        },
                    ],
                    span: Span(3, 9),
                },
                Chain {
                    chain: [
                        Expr {
                            expr: Symbol(
                                Symbol(2: "->"),
                            ),
                            span: Span(10, 12),
                        },
                    ],
                    span: Span(10, 12),
                },
                Chain {
                    chain: [
                        Expr {
                            expr: Bracket {
                                ty: Round,
                                inner: [
                                    Sent {
                                        chains: [
                                            Chain {
                                                chain: [
                                                    Expr {
                                                        expr: Symbol(
                                                            Symbol(3: "T"),
                                                        ),
                                                        span: Span(14, 15),
                                                    },
                                                ],
                                                span: Span(14, 15),
                                            },
                                        ],
                                        span: Span(14, 15),
                                    },
                                ],
                            },
                            span: Span(13, 16),
                        },
                    ],
                    span: Span(13, 16),
                },
            ],
            span: Span(0, 16),
        },
        extension: [
            Line {
                sent: Sent {
                    chains: [
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(0: "fn"),
                                    ),
                                    span: Span(23, 25),
                                },
                            ],
                            span: Span(23, 25),
                        },
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(4: "<--"),
                                    ),
                                    span: Span(26, 29),
                                },
                            ],
                            span: Span(26, 29),
                        },
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(5: "is"),
                                    ),
                                    span: Span(30, 32),
                                },
                            ],
                            span: Span(30, 32),
                        },
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(6: "simple"),
                                    ),
                                    span: Span(33, 39),
                                },
                            ],
                            span: Span(33, 39),
                        },
                    ],
                    span: Span(23, 39),
                },
                extension: [],
                block: [],
                span: Span(23, 39),
            },
        ],
        block: [
            Line {
                sent: Sent {
                    chains: [
                        Chain {
                            chain: [
                                Expr {
                                    expr: Bracket {
                                        ty: Round,
                                        inner: [
                                            Sent {
                                                chains: [
                                                    Chain {
                                                        chain: [
                                                            Expr {
                                                                expr: Bracket {
                                                                    ty: Round,
                                                                    inner: [
                                                                        Sent {
                                                                            chains: [
                                                                                Chain {
                                                                                    chain: [
                                                                                        Expr {
                                                                                            expr: LitStr(
                                                                                                "hello world \t \n \\ \" is ok? ",
                                                                                            ),
                                                                                            span: Span(44, 77),
                                                                                        },
                                                                                        Expr {
                                                                                            expr: Dot,
                                                                                            span: Span(77, 78),
                                                                                        },
                                                                                        Expr {
                                                                                            expr: Symbol(
                                                                                                Symbol(7: "right"),
                                                                                            ),
                                                                                            span: Span(78, 83),
                                                                                        },
                                                                                    ],
                                                                                    span: Span(44, 83),
                                                                                },
                                                                            ],
                                                                            span: Span(44, 83),
                                                                        },
                                                                    ],
                                                                },
                                                                span: Span(43, 84),
                                                            },
                                                        ],
                                                        span: Span(43, 84),
                                                    },
                                                ],
                                                span: Span(43, 84),
                                            },
                                        ],
                                    },
                                    span: Span(42, 85),
                                },
                            ],
                            span: Span(42, 85),
                        },
                    ],
                    span: Span(42, 85),
                },
                extension: [
                    Line {
                        sent: Sent {
                            chains: [
                                Chain {
                                    chain: [
                                        Expr {
                                            expr: Symbol(
                                                Symbol(8: "extend"),
                                            ),
                                            span: Span(94, 100),
                                        },
                                    ],
                                    span: Span(94, 100),
                                },
                                Chain {
                                    chain: [
                                        Expr {
                                            expr: Bracket {
                                                ty: Round,
                                                inner: [
                                                    Sent {
                                                        chains: [
                                                            Chain {
                                                                chain: [
                                                                    Expr {
                                                                        expr: Symbol(
                                                                            Symbol(9: "doing"),
                                                                        ),
                                                                        span: Span(102, 107),
                                                                    },
                                                                ],
                                                                span: Span(102, 107),
                                                            },
                                                            Chain {
                                                                chain: [
                                                                    Expr {
                                                                        expr: Bracket {
                                                                            ty: Square,
                                                                            inner: [
                                                                                Sent {
                                                                                    chains: [
                                                                                        Chain {
                                                                                            chain: [
                                                                                                Expr {
                                                                                                    expr: LitInt(
                                                                                                        1,
                                                                                                    ),
                                                                                                    span: Span(109, 110),
                                                                                                },
                                                                                            ],
                                                                                            span: Span(109, 110),
                                                                                        },
                                                                                        Chain {
                                                                                            chain: [
                                                                                                Expr {
                                                                                                    expr: Symbol(
                                                                                                        Symbol(10: "+"),
                                                                                                    ),
                                                                                                    span: Span(111, 112),
                                                                                                },
                                                                                            ],
                                                                                            span: Span(111, 112),
                                                                                        },
                                                                                        Chain {
                                                                                            chain: [
                                                                                                Expr {
                                                                                                    expr: LitInt(
                                                                                                        2345,
                                                                                                    ),
                                                                                                    span: Span(113, 117),
                                                                                                },
                                                                                            ],
                                                                                            span: Span(113, 117),
                                                                                        },
                                                                                    ],
                                                                                    span: Span(109, 117),
                                                                                },
                                                                            ],
                                                                        },
                                                                        span: Span(108, 118),
                                                                    },
                                                                ],
                                                                span: Span(108, 118),
                                                            },
                                                        ],
                                                        span: Span(102, 118),
                                                    },
                                                ],
                                            },
                                            span: Span(101, 119),
                                        },
                                    ],
                                    span: Span(101, 119),
                                },
                            ],
                            span: Span(94, 119),
                        },
                        extension: [],
                        block: [],
                        span: Span(94, 119),
                    },
                ],
                block: [],
                span: Span(0, 119),
            },
            Line {
                sent: Sent {
                    chains: [
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(11: "return"),
                                    ),
                                    span: Span(122, 128),
                                },
                            ],
                            span: Span(122, 128),
                        },
                    ],
                    span: Span(122, 128),
                },
                extension: [],
                block: [],
                span: Span(122, 128),
            },
        ],
        span: Span(0, 128),
    },
    Line {
        sent: Sent {
            chains: [
                Chain {
                    chain: [
                        Expr {
                            expr: Dot,
                            span: Span(130, 131),
                        },
                        Expr {
                            expr: Symbol(
                                Symbol(0: "fn"),
                            ),
                            span: Span(131, 133),
                        },
                    ],
                    span: Span(130, 133),
                },
                Chain {
                    chain: [
                        Expr {
                            expr: Symbol(
                                Symbol(12: "test_set"),
                            ),
                            span: Span(134, 142),
                        },
                        Expr {
                            expr: Bracket {
                                ty: Round,
                                inner: [],
                            },
                            span: Span(142, 144),
                        },
                    ],
                    span: Span(134, 144),
                },
            ],
            span: Span(130, 144),
        },
        extension: [],
        block: [
            Line {
                sent: Sent {
                    chains: [
                        Chain {
                            chain: [
                                Expr {
                                    expr: Bracket {
                                        ty: Curly,
                                        inner: [
                                            Sent {
                                                chains: [
                                                    Chain {
                                                        chain: [
                                                            Expr {
                                                                expr: Symbol(
                                                                    Symbol(13: "gen"),
                                                                ),
                                                                span: Span(149, 152),
                                                            },
                                                        ],
                                                        span: Span(149, 152),
                                                    },
                                                    Chain {
                                                        chain: [
                                                            Expr {
                                                                expr: LitInt(
                                                                    3,
                                                                ),
                                                                span: Span(153, 154),
                                                            },
                                                        ],
                                                        span: Span(153, 154),
                                                    },
                                                ],
                                                span: Span(149, 154),
                                            },
                                        ],
                                    },
                                    span: Span(147, 156),
                                },
                                Expr {
                                    expr: Dot,
                                    span: Span(156, 157),
                                },
                                Expr {
                                    expr: Symbol(
                                        Symbol(14: "map"),
                                    ),
                                    span: Span(157, 160),
                                },
                                Expr {
                                    expr: Bracket {
                                        ty: Round,
                                        inner: [
                                            Sent {
                                                chains: [
                                                    Chain {
                                                        chain: [
                                                            Expr {
                                                                expr: Symbol(
                                                                    Symbol(15: "?"),
                                                                ),
                                                                span: Span(161, 162),
                                                            },
                                                        ],
                                                        span: Span(161, 162),
                                                    },
                                                ],
                                                span: Span(161, 162),
                                            },
                                        ],
                                    },
                                    span: Span(160, 163),
                                },
                                Expr {
                                    expr: Dot,
                                    span: Span(163, 164),
                                },
                                Expr {
                                    expr: Symbol(
                                        Symbol(16: "fold"),
                                    ),
                                    span: Span(164, 168),
                                },
                                Expr {
                                    expr: Bracket {
                                        ty: Round,
                                        inner: [
                                            Sent {
                                                chains: [
                                                    Chain {
                                                        chain: [
                                                            Expr {
                                                                expr: Symbol(
                                                                    Symbol(10: "+"),
                                                                ),
                                                                span: Span(169, 170),
                                                            },
                                                        ],
                                                        span: Span(169, 170),
                                                    },
                                                ],
                                                span: Span(169, 170),
                                            },
                                        ],
                                    },
                                    span: Span(168, 171),
                                },
                            ],
                            span: Span(147, 171),
                        },
                        Chain {
                            chain: [
                                Expr {
                                    expr: Symbol(
                                        Symbol(15: "?"),
                                    ),
                                    span: Span(172, 173),
                                },
                            ],
                            span: Span(172, 173),
                        },
                    ],
                    span: Span(147, 173),
                },
                extension: [],
                block: [
                    Line {
                        sent: Sent {
                            chains: [
                                Chain {
                                    chain: [
                                        Expr {
                                            expr: Dot,
                                            span: Span(178, 179),
                                        },
                                        Expr {
                                            expr: Symbol(
                                                Symbol(17: "read"),
                                            ),
                                            span: Span(179, 183),
                                        },
                                    ],
                                    span: Span(178, 183),
                                },
                                Chain {
                                    chain: [
                                        Expr {
                                            expr: Dot,
                                            span: Span(184, 185),
                                        },
                                        Expr {
                                            expr: Symbol(
                                                Symbol(18: "int"),
                                            ),
                                            span: Span(185, 188),
                                        },
                                    ],
                                    span: Span(184, 188),
                                },
                            ],
                            span: Span(178, 188),
                        },
                        extension: [],
                        block: [],
                        span: Span(178, 188),
                    },
                ],
                span: Span(0, 188),
            },
        ],
        span: Span(0, 188),
    },
]